app/components/CleanComponent.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+import React, { Component } from "react";
+import {
+  Grid,
+  Button,
+  Icon,
+  Segment,
+  Header,
+  Dropdown
+} from "semantic-ui-react";
+import { isNil } from "lodash";
+import styles from "./styles/common.css";
+import {
+  EXPERIMENTS,
+  DEVICES,
+  KERNEL_STATUS
+} from "../constants/constants";
+import { Kernel } from "../constants/interfaces";
+import { readEEGDataDir } from "../utils/filesystem/write";
+
+interface Props {
```

Comment:
Add kernelStatus prop and refactor some of the conditonals to rely on that instead of the existence of the kernel, etc.